sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/ExpressionConverter.java

```diff
@@ -805,6 +807,22 @@ private RexNode convertSimpleValueToRexNode(TypeKind kind, Value value) {
                 .makeLiteral(
                     value.getStringValue(), typeFactory().createSqlType(SqlTypeName.VARCHAR), true);
         break;
+      case TYPE_NUMERIC:
+        // As ZetaSQL NUMERIC literal would be unparsed as DOUBLE through Calcite makeExactLiteral
```

Comment:
Technically speaking, it is not "unparsed as DOUBLE". Also, I would update the comment to be more descriptive, like:

Cannot simply call makeExactLiteral() because later it will be unparsed to the string representation of the BigDecimal itself (e.g. "SELECT NUMERIC '0'" will be unparsed to "SELECT 0E-9"), and Calcite does not allow customize unparsing of SqlNumericLiteral. So we create a wrapper function here such that we can later recognize it and customize its unparsing in BeamBigQuerySqlDialect.

Reply:
OK. That's make sense.